### ISSUE TITLE
Add aggregated summary to the test results

### ISSUE THREAD

**Issue by PraneshASP** [foundry-rs/foundry#5195]
### Component

Forge

### Describe the feature you would like

Currently the test result (`Test result: ok. 16 passed; 0 failed; finished in 17.27ms`)  is being displayed only for each suite. It would be helpful to display aggregated test results.

### Additional context

Consider there are 4 test suites with 10 tests in each suite. The last line of the output after displaying the test results could be something like:

`Test suites: 4. Total passed: 33. Total failed: 7`

Or it can be more detailed like:

```sh
- VaultTest ~ 10 tests
   - Passed: 6
   - Failed: 4

- VaultFactoryTest ~ 4 tests 
    - Passed: 4
    - Failed: 0
...
Test suites: 2. Total passed: 14. Total failed: 4
```

**Comment by Evalir** [comment]
I like this actually—maybe not as a default option, but behind a flag e.g `--agregate`. wdyt @mattsse?

This would be pretty easy to implement.

**Comment by mds1** [comment]
I'd probably simplify it and just always show the aggregate stats as the last line, it does seem useful and a flag feels unnecessary.

Though if tests fail it seems we currently do show a summary like this:
```
Encountered a total of 1 failing tests, 29 tests succeeded
```

But if tests pass we don't show it. Partly related to #5205 where we'd also want to show number skipped

**Comment by andyrobert3** [comment]
i think this should be straightforward, i am interested on taking it!

@Evalir

**Comment by Evalir** [comment]
For sure! Go for it!

Note that #5205 might introduce conflicts

**Comment by andyrobert3** [comment]
should be done this weekend!

**Comment by andyrobert3** [comment]
https://github.com/foundry-rs/foundry/pull/5266

@Evalir @mds1 

Welcome feedback on how we can better present the test results to the user!

**Comment by andyrobert3** [comment]
issue closed with https://github.com/foundry-rs/foundry/pull/5266

**Comment by Evalir** [comment]
Closed with #5266

**Comment by PraneshASP** [comment]
Thank you, shipooors!  🫡
